lib/foreman/logging.rb

```diff
@@ -98,9 +99,11 @@ def exception(context_message, exception, options = {})
       }
       extra_fields[:foreman_code] = exception.code if exception.respond_to?(:code)
       with_fields(extra_fields) do
-        self.logger(logger_name).public_send(level) do
-          ([context_message, "#{exception.class}: #{exception.message}"] + backtrace).join("\n")
-        end
+        self.logger(logger_name).public_send(level) { context_message }
+      end
+      # backtrace have its own separate level to prevent flooding logs with backtraces
+      self.logger(logger_name).public_send(backtrace_level) do
+        "Backtrace for '#{context_message}' error (#{exception.class}): #{exception.message}\n" + backtrace.join("\n")
       end
     end
```

Comment:
And this method is covered in many tests, we do actually log during unit/integration testing into `log/test.log`.